cmd/http-check/main.go

```diff
@@ -30,29 +31,52 @@ func init() {
 
 func main() {
 	log.Infoln("Beginning check.")
+	checksRan := 0
+	checksPassed := 0
+	checksFailed := 0
+
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+	// This for loop makes a http GET request to a known internet address, address can be changed in deployment spec yaml
+	// and returns a http status every second.
+	for checksRan < 10 {
+		<-ticker.C
+		r, err := http.Get(checkURL)
+		checksRan++
 
-	// Make a GET request.
-	r, err := http.Get(checkURL)
-	if err != nil {
-		reportErr := fmt.Errorf("error occurred performing a " + http.MethodGet + " to " + checkURL + ": " + err.Error())
-		log.Errorln(reportErr.Error())
-		err = kh.ReportFailure([]string{reportErr.Error()})
 		if err != nil {
-			log.Fatalln("error when reporting to kuberhealthy:", err.Error())
+			checksFailed++
+			log.Infoln("Failed to reach URL: ", checkURL)
+			continue
 		}
-		os.Exit(0)
-	}
 
-	// Check if the response status code is a 200.
-	if r.StatusCode == http.StatusOK {
+		if r.StatusCode != http.StatusOK {
+			log.Infoln("Got a", r.StatusCode, "with a", http.MethodGet, "to", checkURL)
+			checksFailed++
+			continue
+		}
 		log.Infoln("Got a", r.StatusCode, "with a", http.MethodGet, "to", checkURL)
-		kh.ReportSuccess()
-		os.Exit(0)
+		checksPassed++
 	}
 
-	reportErr := fmt.Errorf("unable to retrieve a " + strconv.Itoa(http.StatusOK) + " from " + checkURL + " got a " + strconv.Itoa(r.StatusCode) + " instead")
-	log.Errorln(reportErr.Error())
-	kh.ReportFailure([]string{reportErr.Error()})
+	// Displays the results of 10 URL requests
+	log.Infoln(checksRan, "checks ran")
+	log.Infoln(checksPassed, "checks passed")
+	log.Infoln(checksFailed, "checks failed")
+
+	// Check to see if the 10 requests passed at 80% and reports to Kuberhealthy accordingly
+	if checksPassed < 8 {
+		reportErr := fmt.Errorf("unable to retrieve a http.StatusOK from " + checkURL + "check failed " + strconv.Itoa(checksFailed) + " out of 10 attempts")
```

Comment:
make sure to move the htt.StatusOK out of the first string -- it's a constant from the http package that should be put in the error

Reply:
changed verbiage to "unable to retrieve a valid response from " as the checks failed is more than just http.StatusOk @jonnydawg